File: python/colors/colors.py
FOREGROUND = 38
BACKGROUND = 48
INITIAL = 0
FINAL = 255

def color_range(start, end, anyground):
    resetdefault = "\x1b[0m"
    resetwithspace = "\x1b[0m "

    resetoutput = resetwithspace
    for color in range(start, end+1):
        if color % 16 == 0:
            print()

        if color == end:
            resetoutput = resetdefault

        print("\x1b[%d;5;%dm%3d%s" % (anyground, color, color, resetoutput), end='')


def main():
    print(f"\x1b[1;39m{INITIAL} to {FINAL} Colors (FOREGROUND)\x1b[22;0m")
    color_range(INITIAL, FINAL, FOREGROUND)
    print("\n")
    print(f"\x1b[1;39m{INITIAL} to {FINAL} Colors (BACKGROUND)\x1b[22;0m")
    color_range(INITIAL, FINAL, BACKGROUND)
    print("\n")

if __name__ == '__main__':
    main()

File: js/colors/colors.js
const FOREGROUND = 38;
const BACKGROUND = 48;
const INITIAL = 0;
const FINAL = 255;

function color_range(start, end, anyground) {

    const resetdefault = "\x1b[0m";
    const resetwithspace = "\x1b[0m ";

    let resetoutput = resetwithspace;

    for (let color = start; color <= end; color++) {
        if ((color % 16) === 0) {
            console.log();
        }

        if (color == end) {
            resetoutput = resetdefault;
        }

        process.stdout.write(`\x1b[${anyground};5;${color}m${color.toString().padStart(3)}${resetoutput}`);
    }
}

process.stdout.write(`\x1b[1;39m${INITIAL} to ${FINAL} Colors (FOREGROUND)\x1b[22;0m\n`);
color_range(INITIAL, FINAL, FOREGROUND);
console.log("\n");

process.stdout.write(`\x1b[1;39m${INITIAL} to ${FINAL} Colors (BACKGROUND)\x1b[22;0m\n`);
color_range(INITIAL, FINAL, BACKGROUND);
console.log("\n");
